lib/services/local/authinfo.go

```diff
@@ -0,0 +1,111 @@
+/*
+ * Teleport
+ * Copyright (C) 2025  Gravitational, Inc.
+ *
+ * This program is free software: you can redistribute it and/or modify
+ * it under the terms of the GNU Affero General Public License as published by
+ * the Free Software Foundation, either version 3 of the License, or
+ * (at your option) any later version.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU Affero General Public License for more details.
+ *
+ * You should have received a copy of the GNU Affero General Public License
+ * along with this program.  If not, see <http://www.gnu.org/licenses/>.
+ */
+
+package local
+
+import (
+	"context"
+
+	"github.com/coreos/go-semver/semver"
+	"github.com/gravitational/trace"
+
+	"github.com/gravitational/teleport/api/gen/proto/go/teleport/authinfo/v1"
+	"github.com/gravitational/teleport/api/types"
+	authinfotype "github.com/gravitational/teleport/api/types/authinfo"
+	"github.com/gravitational/teleport/lib/backend"
+	"github.com/gravitational/teleport/lib/services"
+)
+
+const (
+	// authInfoPrefix is a backend key for storing the auth server information.
+	authInfoPrefix = "auth_info"
+)
+
+// AuthInfoService is responsible for managing the information about auth server.
+type AuthInfoService struct {
+	backed backend.Backend
+}
+
+// NewAuthInfoService returns a new AuthInfoService.
+func NewAuthInfoService(b backend.Backend) (*AuthInfoService, error) {
+	return &AuthInfoService{
+		backed: b,
+	}, nil
+}
+
+// GetTeleportVersion reads the last known Teleport version from storage.
+func (s *AuthInfoService) GetTeleportVersion(ctx context.Context) (*semver.Version, error) {
+	item, err := s.backed.Get(ctx, backend.NewKey(authInfoPrefix))
+	if err != nil {
+		return nil, trace.Wrap(err)
+	}
+
+	info, err := services.UnmarshalProtoResource[*authinfo.AuthInfo](item.Value)
+	if err != nil {
+		return nil, trace.Wrap(err)
+	}
+	if err := authinfotype.ValidateAuthInfo(info); err != nil {
+		return nil, trace.Wrap(err)
+	}
+	version, err := semver.NewVersion(info.GetSpec().TeleportVersion)
+	if err != nil {
+		return nil, trace.Wrap(err)
+	}
+
+	return version, nil
+}
+
+// WriteTeleportVersion writes the last known Teleport version to the storage.
+func (s *AuthInfoService) WriteTeleportVersion(ctx context.Context, version *semver.Version) error {
+	if version == nil {
```

Comment:
I think we should stick with RFD 156 and expose the full Create Update Read Delete operations. Each time we've done differently we ended up needs every operation at some point.

Reply:
I don't know if it's necessary but it wouldn't hurt.